core/src/main/java/tc/oc/pgm/loot/LootableMatchModule.java

```diff
@@ -0,0 +1,337 @@
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.filters.FilterMatchModule;
+import tc.oc.pgm.filters.query.BlockQuery;
+import tc.oc.pgm.filters.query.EntityQuery;
+import tc.oc.pgm.itemmeta.ItemModifyMatchModule;
+import tc.oc.pgm.kits.Slot;
+import tc.oc.pgm.util.collection.DefaultMapAdapter;
+import tc.oc.pgm.util.collection.InstantMap;
+
+/**
+ * After {@link MapProtos#LOOTABLE_FILLER_CHANGES} fillers with dynamic triggers no longer trigger
+ * at the start of the match unless explicitly specified. In addition if a filler is triggered
+ * multiple times before a chest is opened it will be filled with that filler the amount of times
+ * the filler was triggered, instead of just once.
+ *
+ * <p>For relevant places in code look for the {@link #triggerFillerChanges} usages in this class
+ */
+@ListenerScope(value = MatchScope.LOADED)
+public class LootableMatchModule implements MatchModule, Listener {
+
+  private final Logger logger;
+  private final Match match;
+  private final List<FillerDefinition> fillers;
+  private final List<Cache> caches;
+
+  private final InstantMap<FillableAndFiller> filledAt;
+  private final Map<FillerDefinition, Integer> triggeredFillers;
+
+  private final Table<Fillable, FillerDefinition, Integer> triggerHistory;
+
+  private final boolean triggerFillerChanges; // See javadoc
+  private final Consumer<FillerDefinition> triggeredFillerInvalidator;
+
+  private final @Nullable ItemModifyMatchModule immm;
+
+  public LootableMatchModule(
+      Logger logger, Match match, List<FillerDefinition> fillers, List<Cache> caches) {
+    this.logger = logger;
+    this.fillers = fillers;
+    this.match = match;
+    this.caches = caches;
+    this.filledAt = new InstantMap<>(new WorldTickClock(match.getWorld()));
+    this.triggeredFillers = new DefaultMapAdapter<>(0, fillers.size(), true);
+    this.triggerHistory = HashBasedTable.create();
+    this.triggerFillerChanges =
+        match.getMap().getProto().isNoOlderThan(MapProtos.LOOTABLE_FILLER_CHANGES);
+    this.triggeredFillerInvalidator =
+        this.triggerFillerChanges
+            ? filler -> this.triggeredFillers.compute(filler, (fill, count) -> ++count)
+            : filler -> this.filledAt.keySet().removeIf(f -> filler.equals(f.getFiller()));
+    this.immm = match.getModule(ItemModifyMatchModule.class);
+
+    fillers.forEach(
+        filler -> {
+          if (filler.getRefillTrigger() == null) return;
+          match
+              .needModule(FilterMatchModule.class)
+              .onRise(
+                  Match.class,
+                  filler.getRefillTrigger(),
+                  m -> this.triggeredFillerInvalidator.accept(filler));
+        });
+  }
+
+  /**
+   * Return a predicate that applies a Filter to the given InventoryHolder, or null if the
+   * InventoryHolder is not something that we should be filling.
+   */
+  private static @Nullable Predicate<Filter> passesFilter(InventoryHolder holder) {
```

Comment:
```suggestion
  private static @Nullable Predicate<Filter> getFilterPredicate(InventoryHolder holder) {
```
the wording `passesFilter` implies you're checking the filter, when you clearly aren't. You're just creating a predicate to see if it passes the filter